packages/front-end/components/Experiment/AnalysisSettingsBar.tsx

```diff
@@ -313,22 +314,49 @@ export default function AnalysisSettingsBar({
                   <form
                     onSubmit={(e) => {
                       e.preventDefault();
-                      apiCall(`/experiment/${experiment.id}/snapshot`, {
-                        method: "POST",
-                        body: JSON.stringify({
-                          phase,
-                          dimension,
-                          statsEngine,
-                          regressionAdjustmentEnabled,
-                          metricRegressionAdjustmentStatuses,
-                        }),
-                      })
-                        .then(() => {
+                      const snapshotProps: Omit<TrackSnapshotProps, "id"> = {
+                        source: "RunQueriesButton",
+                        experiment: experiment.id,
+                        engine: statsEngine,
+                        datasource_type: datasource?.type || null,
+                        regression_adjustment_enabled: !!regressionAdjustmentEnabled,
+                        sequential_testing_enabled: !!experiment.sequentialTestingEnabled,
+                        sequential_testing_tuning_parameter:
+                          experiment.sequentialTestingTuningParameter,
+                        skip_partial_data: !!experiment.skipPartialData,
+                        activation_metric_selected: !!experiment.activationMetric,
+                        query_filter_selected: !!experiment.queryFilter,
+                        segment_selected: !!experiment.segment,
+                        dimension: dimension || "",
+                      };
+                      apiCall<{ snapshot: ExperimentSnapshotInterface }>(
+                        `/experiment/${experiment.id}/snapshot`,
+                        {
+                          method: "POST",
+                          body: JSON.stringify({
+                            phase,
+                            dimension,
+                            statsEngine,
+                            regressionAdjustmentEnabled,
+                            metricRegressionAdjustmentStatuses,
+                          }),
+                        }
+                      )
+                        .then((res) => {
+                          trackSnapshot("create", {
+                            ...snapshotProps,
+                            id: res.snapshot.id,
+                          });
                           mutate();
                           setRefreshError("");
                         })
                         .catch((e) => {
                           setRefreshError(e.message);
+                          trackSnapshot("error", {
+                            ...snapshotProps,
+                            id: "",
+                            error: e.message,
+                          });
```

Comment:
This PR can be drastically simplified with some small tweaks.

The back-end is already doing the logic to create a snapshot from an experiment/dimension/statsEngine/CUPED/overrides combo.  This logic is complicated and likely to change in the future, so instead of replicating it on the front-end, we can just use the fully built `SnapshotInterface`, which is returned from the back-end as `res.snapshot`.

We will need some logic to go from `SnapshotInterface` or `ReportInterface` to `TrackSnapshotProps`, but that should be much simpler and it will be identical no matter where the tracking call is made from.  So we can just make 2 utility functions - `getTrackingPropsFromSnapshot` and `getTrackingPropsFromReport`.

The last part you need to handle is the catch block.  Unfortunately, we don't have the fully built snapshot from the API response, but I think that's ok.  Snapshot creation almost never fails.  It queues up a background job and returns immediately, so query errors won't be caught here.  So we can either just remove tracking from the catch block entirely, or simplify it to just track the error message and no other snapshot properties (since it's likely just a random Mongo error and not anything related to the snapshot itself).

The same comments apply to all the usages below as well.  Doing this I think will bring the PR from 400+ lines down closer to 100 and will be more maintainable (less likely to break as we make further changes to snapshots)

Reply:
> The last part you need to handle is the catch block. Unfortunately, we don't have the fully built snapshot from the API response, but I think that's ok. 

This was the main reason I didn't go to just passing the `SnapshotInterface` when I saw how unnecessarily long and complex a lot of this PR was.

> (since it's likely just a random Mongo error and not anything related to the snapshot itself).

Yeah this was my sense. I feel like tracking failing queries is another thing we should do.

> simplify it to just track the error message and no other snapshot properties

I'm going to go this route.

Reply:
Decided to go with removing errors altogether since they aren't the main ones of interest.

I also kept passing `datasourceType` because I wasn't 100% sure on the context needed in `track.ts` to generate it at that point. Please advise if there's a way to do this.